main/lsp/LSPTypechecker.cc

```diff
@@ -246,28 +246,40 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
         fref.data(*gs).strictLevel = pipeline::decideStrictLevel(*gs, fref, config->opts);
 
         toTypecheck.emplace_back(fref);
+    }
+
+    UnorderedSet<core::FileRef> packageFiles;
 
+    for (auto fref : toTypecheck) {
         // Only need to re-run packager if we're going to delete constants and have to re-define
         // their visibility, which only happens if we're running incrementalNamer.
         if (shouldRunIncrementalNamer) {
             // NOTE: Using `gs` to access package information here assumes that edits to __package.rb
             // files don't take the fast path. We'll want (or maybe need) to revisit this when we start
             // making edits to `__package.rb` take fast paths.
             if (!(fref.data(*gs).isPackage())) {
-                auto pkgName = gs->packageDB().getPackageNameForFile(fref);
-                if (pkgName.exists()) {
+                auto &pkg = gs->packageDB().getPackageForFile(*gs, fref);
+                if (pkg.exists()) {
                     // Since even no-op (e.g. whitespace-only) edits will cause constants to be deleted
                     // and re-added, we have to add the __package.rb files to set of files to retypecheck
                     // so that we can re-run PropagateVisibility to set export bits for any constants.
-                    auto packageFref = gs->packageDB().getPackageInfo(pkgName).fullLoc().file();
-                    if (result.changedFiles.find(packageFref) == result.changedFiles.end()) {
-                        // Skip duplicates
-                        toTypecheck.emplace_back(packageFref);
+                    auto packageFref = pkg.fullLoc().file();
+                    if (!packageFref.exists()) {
+                        continue;
                     }
+
+                    packageFiles.emplace(packageFref);
                 }
             }
         }
     }
+
+    for (auto packageFref : std::move(packageFiles)) {
```

Comment:
I think this `move` is unnecessary.  (I guess it's trying to say that we're done with `packageFiles`, which I agree is helpful information...but I don't think that fact is relevant here.)

Reply:
I agree -- but I think I wanted to leave it there to essentially communicate that to the reader or future editors of this code. It was helpful, for example, on line 220 to have `std::move(result.extraFiles)`, so I knew I couldn't use it down here.